pkg/api/customization/oci/listers.go

```
@@ -0,0 +1,257 @@
+	"net/http"
+	"strings"
+
+	"github.com/oracle/oci-go-sdk/common"
+	"github.com/oracle/oci-go-sdk/containerengine"
+	"github.com/oracle/oci-go-sdk/core"
+	"github.com/oracle/oci-go-sdk/identity"
+	"github.com/rancher/norman/httperror"
+	"github.com/sirupsen/logrus"
+)
+
+func processVcns(provider common.ConfigurationProvider, compartment string) ([]byte, int, error) {
+	logrus.Debugf("[oci-handler] listing VCNs in compartment: %s", compartment)
+	virtualNetworkClient, err := core.NewVirtualNetworkClientWithConfigurationProvider(provider)
+	if err != nil {
+		logrus.Debugf("[oci-handler] error creating OCI Virtual Network client: %v", err)
+		return nil, httperror.ServerError.Status, err
+	}
+	vcnRequest := core.ListVcnsRequest{
+		CompartmentId: &compartment,
+	}
+	vcnResponse, err := virtualNetworkClient.ListVcns(context.Background(), vcnRequest)
+	if err != nil {
+		httpErr := httperror.ErrorCode{}
+		if vcnResponse.RawResponse != nil {
+			httpErr.Status = vcnResponse.RawResponse.StatusCode
+			httpErr.Code = vcnResponse.RawResponse.Status
+		} else {
+			httpErr.Status = httperror.ServerError.Status
+			httpErr.Code = "error listing VCNs"
+		}
+		logrus.Debugf("[oci-handler] error listing VCNs with OCI Virtual Network client: %v", err)
+		return nil, httpErr.Status, err
+	}
+
+	var vcnDisplayNames []string
+	for _, item := range vcnResponse.Items {
+		vcnDisplayNames = append(vcnDisplayNames, *(item.DisplayName))
+	}
+
+	data, err := json.Marshal(vcnDisplayNames)
+	if err != nil {
+		return data, httperror.ServerError.Status, err
+	}
+
+	return data, http.StatusOK, err
+}
+
+func processOkeVersions(provider common.ConfigurationProvider) ([]byte, int, error) {
+	logrus.Debug("[oci-handler] listing OKE versions")
+	containerClient, err := containerengine.NewContainerEngineClientWithConfigurationProvider(provider)
+	if err != nil {
+		logrus.Debugf("[oci-handler] error creating OCI ContainerEngine client: %v", err)
+		return nil, http.StatusOK, err
```

This is an error state, it should return an error